sixpack.js

```diff
@@ -199,42 +196,37 @@
         } else {
             var httpModule = url.startsWith('https') ? 'https' : 'http';
             var http = eval('require')(httpModule); // using eval to skip webpack bundling and warnings
-
             var parsedUrl = new URL(url);
             var options = {
                 protocol: parsedUrl.protocol,
                 port: parsedUrl.port,
                 hostname: parsedUrl.hostname,
                 path: parsedUrl.pathname + parsedUrl.search, 
-                headers: { Cookie: cookie }
+                headers: { Cookie: cookie },
+                timeout,
             }
             var req = http.get(options, function(res) {
                 var body = "";
                 res.on('data', function(chunk) {
                     return body += chunk;
                 });
                 return res.on('end', function() {
-                    var data = { status: 'failed', response: body };
-
                     if (res.statusCode < 500) {
                         try {
-                            data = JSON.parse(body);
+                            return callback(null, JSON.parse(body));
                         } catch (err) {
-                            console.error(err);
+                            console.error('error parsing the request body', err);
                         }
                     }
-
-                    if (!timed_out) {
-                        clearTimeout(timeout_handle);
-                        return callback(null, data);
-                    }
+                    return callback(null, { status: 'failed', response: body });
                 });
             });
             req.on('error', function(err) {
-                if (!timed_out) {
-                    clearTimeout(timeout_handle);
-                    return callback(err);
-                }
+                return callback(err);
+            });
+            req.on('timeout', function() {
+                console.error('timeout while requesting: ' + url);
+                req.abort();
```

Comment:
As it will be running inside all our JS projects, should we add something to identify this log later? 🤔 
Something like this, maybe?
```suggestion
                console.error('[sixpack-js] timeout while requesting: ' + url);
                req.abort();
```

Reply:
done